File: php/features.php
<?php
declare(strict_types=1);

// CatherineSchulmanQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CatherineSchulmanQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CatherineSchulmanQuotesBaseFeature();
            case "test":
                return new CatherineSchulmanQuotesTestFeature();
            default:
                return new CatherineSchulmanQuotesBaseFeature();
        }
    }
}

File: py/features.py
# CatherineSchulmanQuotes SDK feature factory

from feature.base_feature import CatherineSchulmanQuotesBaseFeature
from feature.test_feature import CatherineSchulmanQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CatherineSchulmanQuotesBaseFeature(),
        "test": lambda: CatherineSchulmanQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
